### PR TITLE
Add nrf5340-app-hal

### PR DESCRIPTION
**Pull request by sphw**
This PR adds a HAL for the nRF5340's app core. It is more or less identical to the nRF9160 HAL that already exists, with some features added or removed for the nRF5340. I have not added support for the net core yet, as I don't need it for my use case.

### CODE REVIEW

File: nrf5340-app-hal/log.log [hunk 1]
```diff
@@ -0,0 +1,3527 @@
+[2021-Dec-03 05:29:48] [ info] --------------------------------------------------------------------------------
```
Comment (jonas-schievink) [inline nrf5340-app-hal/log.log:1]: Looks like this was checked in by accident

Comment (sphw) [reply nrf5340-app-hal/log.log]: Oh yup. Just removed it

**Comment by jonas-schievink** [comment]
Please add the new HAL to the array in `xtask/src/lib.rs` to enable CI for it

**Comment by sphw** [comment]
> Please add the new HAL to the array in `xtask/src/lib.rs` to enable CI for it

Done

**Comment by jonas-schievink** [comment]
Thanks, looks pretty good!

bors r+

**Pull request merged**